packages/web-app/src/controllers/valid-appeal-details.js

```diff
@@ -0,0 +1,52 @@
+const {
+  reviewAppealSubmission: previousPage,
+  validAppealDetails: currentPage,
+  home: nextPage,
+} = require('../config/views');
+const saveAndContinue = require('../lib/save-and-continue');
+
+const viewData = (appealId, horizonId, validAppealDetails) => ({
+  pageTitle: 'Valid appeal details',
+  backLink: `/${previousPage}/${appealId}`,
+  validAppealDetails,
+  appealReference: horizonId,
+});
+
+const getValidAppealDetails = (req, res) => {
+  const {
+    session: {
+      appeal: {
+        appeal: { id, horizonId },
+        casework: { validAppealDetails },
+      },
+    },
+  } = req;
+
+  res.render(currentPage, viewData(id, horizonId, validAppealDetails));
+};
+
+const postValidAppealDetails = (req, res) => {
+  const {
+    session: {
+      appeal: {
+        appeal: { id, horizonId },
+      },
+    },
+    body,
+  } = req;
+
+  req.session.appeal.casework.validAppealDetails = req.body['valid-appeal-details'];
+
+  saveAndContinue({
```

Comment:
Is seperating the save and continue functionality to declutter controller, or is it for reuseablity?